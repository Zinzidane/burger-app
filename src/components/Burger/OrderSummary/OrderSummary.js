import React from 'react';
import Aux from '../../../hoc/Auxilary';
import Button from '../../UI/Button/Button';

const orderSummary = (props) => {
  const ingredientSummary = Object.keys(props.ingredients)
    .map(ingKey => {
      return (
        <li key={ingKey}>
          <span style={{textTransform: 'capitalize'}}>{ingKey}</span>: 
          {props.ingredients[ingKey]}
        </li>
      );
    });

  return (
    <Aux>
      <h3>Your order</h3>
      <p>A delicious burger with following ingredients:</p>
      <ul>
        {ingredientSummary}
      </ul>
      <p>Continue to checkout?</p>
      <Button btnType="Danger" clicked={props.purchaseCancelled}>CANCEL</Button>
      <Button btnType="Success" clicked={props.purchaseContinued}>CONTINUE</Button>
    </Aux>
  );
};

export default orderSummary;