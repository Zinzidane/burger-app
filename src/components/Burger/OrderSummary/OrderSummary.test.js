import React from 'react';
import OrderSummary from './OrderSummary';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<OrderSummary />', () => {
  let wrapper;

  const props = {
    ingredients: [],
    price: 0
  };
  beforeEach(() => {
    wrapper = shallow(<OrderSummary {...props} />);
  })

  it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  });
  
});