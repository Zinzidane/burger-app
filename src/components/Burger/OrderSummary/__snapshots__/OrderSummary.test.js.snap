// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderSummary /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <orderSummary
    ingredients={Array []}
    price={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3>
          Your Order
</h3>,
        <p>
          A delicious burger with the following ingredients:
</p>,
        <ul>
          
</ul>,
        <p>
          <strong>
                    Total Price: 
                    0.00
          </strong>
</p>,
        <p>
          Continue to Checkout?
</p>,
        <button
          btnType="Danger"
          clicked={undefined}
>
          CANCEL
</button>,
        <button
          btnType="Success"
          clicked={undefined}
>
          CONTINUE
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Your Order",
        },
        "ref": null,
        "rendered": "Your Order",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A delicious burger with the following ingredients:",
        },
        "ref": null,
        "rendered": "A delicious burger with the following ingredients:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <strong>
            Total Price: 
            0.00
</strong>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Total Price: ",
              "0.00",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Total Price: ",
            "0.00",
          ],
          "type": "strong",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Continue to Checkout?",
        },
        "ref": null,
        "rendered": "Continue to Checkout?",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "btnType": "Danger",
          "children": "CANCEL",
          "clicked": undefined,
        },
        "ref": null,
        "rendered": "CANCEL",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "btnType": "Success",
          "children": "CONTINUE",
          "clicked": undefined,
        },
        "ref": null,
        "rendered": "CONTINUE",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3>
            Your Order
</h3>,
          <p>
            A delicious burger with the following ingredients:
</p>,
          <ul>
            
</ul>,
          <p>
            <strong>
                        Total Price: 
                        0.00
            </strong>
</p>,
          <p>
            Continue to Checkout?
</p>,
          <button
            btnType="Danger"
            clicked={undefined}
>
            CANCEL
</button>,
          <button
            btnType="Success"
            clicked={undefined}
>
            CONTINUE
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your Order",
          },
          "ref": null,
          "rendered": "Your Order",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A delicious burger with the following ingredients:",
          },
          "ref": null,
          "rendered": "A delicious burger with the following ingredients:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              Total Price: 
              0.00
</strong>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Total Price: ",
                "0.00",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Total Price: ",
              "0.00",
            ],
            "type": "strong",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Continue to Checkout?",
          },
          "ref": null,
          "rendered": "Continue to Checkout?",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "btnType": "Danger",
            "children": "CANCEL",
            "clicked": undefined,
          },
          "ref": null,
          "rendered": "CANCEL",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "btnType": "Success",
            "children": "CONTINUE",
            "clicked": undefined,
          },
          "ref": null,
          "rendered": "CONTINUE",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
